src/components/Portal.svelte

```diff
@@ -0,0 +1,25 @@
+<script>
+  /* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+import { onMount, onDestroy } from "svelte";
```

Comment:
a "portal" component will render its children _outside_ of the DOM tree where it's placed. This is a nice way to ensure that a modal, for instance, is always the last item to be rendered.